apps/ledger-live-mobile/src/screens/Market/index.tsx

```diff
@@ -85,14 +85,8 @@ const BottomSection = ({ navigation }: { navigation: NavigationProps["navigation
   const { range, orderBy, order, top100 } = requestParams;
   const starredMarketCoins: string[] = useSelector(starredMarketCoinsSelector);
   const filterByStarredAccount: boolean = useSelector(marketFilterByStarredAccountsSelector);
-  const firstMount = useRef(true); // To known if this is the first mount of the page
 
   useEffect(() => {
-    if (firstMount.current) {
-      // We don't want to refresh the market data directly on mount, the data is already refreshed with wanted parameters from MarketDataProviderWrapper
-      firstMount.current = false;
-      return;
-    }
```

Comment:
following down the rabbit hole =>

i imagine this was made to prevent the re-rendering "cascading" problem. there may be a wider problem to solve here: the fact we do an action depending on parameters -> it sounds like these parameters should instead be fed into the 'useMarketData()'  function (which don't take param) because it's dependency of them, there is probably an antipattern to dig later on this topic.

a typical example of something that shouldn't be an effect https://react.dev/learn/you-might-not-need-an-effect

Reply:
The problem is that the data is not always up to date when it is first rendered. What's more, there's no intenpestive rerender when you remove the condition